ante/README.md

```diff
@@ -0,0 +1,148 @@
+# Gaia Fee Ante-handler
+
+## Gaia Fees
+
+In cosmoshub, there are two types of fees:
+- global fees
+- min_gas_pricess
+  
+### Global Fees
+Global fees are [`sdk.DecCoins`](https://github.com/cosmos/cosmos-sdk/blob/a1777a87b65fad74732cfe1a4c27683dcffffbfe/types/dec_coin.go#L158) type and globally valid. Global fees are set up through gov proposal. [Valid global fees](https://github.com/cosmos/gaia/blob/82c4353ab1b04cf656a8c95d226c30c7845f157b/x/globalfee/types/params.go#L54-L99) have to be and sorted by denom, have have nonnegtive amount, with a valid and unique denomination(denom) (i.e no duplicates), Global fees allow zero coins! Zero coins can help define the desired fee denoms even the chain might not charge fees. Each transaction (except bypass transaction) has to meet the global fees' requirement:
+ - all of paidfees' denoms (except zero coins) have to be a subset of the globalfee's denom set.
+- paidfees contain at least one denom that is present and greater/equal to the amount of the same denom in globalfee.
+
+### query global fees
+You can query globalfee by 
+```shell
+gaiad q globalfee minimum-gas-prices
+# or
+gaiad q params subspace globalfee MinimumGasPricesParam
+```
+### empty global fees  and default global fees
+When global fee is not setup, the query will return empty globalfee `minimum_gas_prices: []`. Gaiad will use `0uatom` as global fee in this case. This is due to the Cosmoshub as default only accepts uatom as fee denom.
+
+### setup global fees
+Global fee can be setup by [gov proposals](https://hub.cosmos.network/main/governance/proposals/).
+```shell
+gov submit-legacy-proposal param-change proposal.json
+````
+A `proposal.json` example:
+```
+{
+  "title": "Global fees Param Change",
+  "description": "Update global fees",
+  "changes": [
+    {
+      "subspace": "globalfee",
+      "key": "MinimumGasPricesParam",
+      "value": [{"denom":"stake", "amount":"0.002"}, {"denom":"uatom", "amount": "0.001"}]
+    }
+  ],
+  "deposit": "1000stake"
+}
+```
+please note: in the above "value" field, coins must sorted by denom.
```

Comment:
Do the denoms have to be alphabetically sorted?

Reply:
yes